GDSCommon-Demo/GDSCommon-DemoTests/GDSInformationControllerTests.swift

```diff
@@ -0,0 +1,171 @@
+    let footnote: GDSLocalisedString? = "Information screen footnote"
+    let primaryButtonViewModel: ButtonViewModel
+    let secondaryButtonViewModel: ButtonViewModel?
+    
+    let rightBarButtonTitle: GDSLocalisedString? = "right bar button"
+    let backButtonIsHidden: Bool = false
+    let appearAction: () -> Void
+    let dismissAction: () -> Void
+    
+    init(primaryButtonAction: @escaping () -> Void,
+         secondaryButtonAction: @escaping () -> Void,
+         appearAction: @escaping () -> Void,
+         dismissAction: @escaping () -> Void
+    ) {
+        primaryButtonViewModel = MockButtonViewModel(title: "Information primary button title") {
+            primaryButtonAction()
+        }
+        secondaryButtonViewModel = MockButtonViewModel(title: "Information secondary button title") {
+            secondaryButtonAction()
+        }
+        self.appearAction = appearAction
+        self.dismissAction = dismissAction
+    }
+    
+    func didAppear() {
+        appearAction()
+    }
+    
+    func didDismiss() {
+        dismissAction()
+    }
+}
+
+extension GDSInformationViewControllerTests {
+    func test_labelContents() throws {
+        XCTAssertNotNil(try sut.informationImage)
+        XCTAssertEqual(try sut.informationImage.tintColor, .gdsPrimary)
+        XCTAssertEqual(try sut.informationTitleLabel.text, "Information screen title")
+        XCTAssertEqual(try sut.informationTitleLabel.font, .largeTitleBold)
+        XCTAssertTrue(try sut.informationTitleLabel.accessibilityTraits.contains(.header))
+        XCTAssertEqual(try sut.informationBodyLabel.text, "Information screen body")
+        XCTAssertFalse(try sut.informationBodyLabel.accessibilityTraits.contains(.header))
+        XCTAssertEqual(try sut.informationFootnoteLabel.text, "Information screen footnote")
+        XCTAssertEqual(try sut.informationFootnoteLabel.font, .footnote)
+        XCTAssertEqual(try sut.informationFootnoteLabel.maximumContentSizeCategory, .accessibilityMedium)
+        XCTAssertFalse(try sut.informationFootnoteLabel.accessibilityTraits.contains(.header))
+        XCTAssertEqual(try sut.informationPrimaryButton.title(for: .normal), "Information primary button title")
+        XCTAssertEqual(try sut.informationSecondaryButton.title(for: .normal), "Information secondary button title")
+    }
+    
+    func test_primaryButtonAction() throws {
+        XCTAssertFalse(primaryButton)
+        try sut.informationPrimaryButton.sendActions(for: .touchUpInside)
+        XCTAssertTrue(primaryButton)
+    }
+    
+    func test_secondaryButtonAction() throws {
+        XCTAssertFalse(secondaryButton)
+        try sut.informationSecondaryButton.sendActions(for: .touchUpInside)
+        XCTAssertTrue(secondaryButton)
+    }
+    
+    func test_didAppear() throws {
+        XCTAssertFalse(viewDidAppear)
+        sut.beginAppearanceTransition(true, animated: false)
+        sut.viewDidAppear(false)
```

Comment:
I don't think we should be calling these lifecycle methods directly if we can avoid them, the `.beginAppearanceTransition(_:, animated:)` and `.endAppearanceTransition()` should be sufficient for managing the screens displaying programmatically.
Maybe take them out and see how the tests perform 🙂